### ISSUE TITLE
hint_pattern not playing well with cmdheight=0

### ISSUE THREAD

**Issue by glyh** [phaazon/hop.nvim#296]
Here's a screenshot:
![image](https://user-images.githubusercontent.com/12870742/181695833-0d610130-1555-4a1b-afd6-e4606bdda55f.png)

You can notice the cursor is on the status bar, which is not as intended.

**Comment by megalithic** [comment]
basically it's got you on the cmd line to type your hint chars. you might be able to do an override where you set cmdheight to 1 then have an after hint commit hook that will set it back to 0. not sure if this all exists, but it seems logical with all the hop.nvim extensions and other ways to extend it.

**Comment by megalithic** [comment]
looks like nvim-surround author got around it with : https://github.com/kylechui/nvim-surround/blob/main/lua/nvim-surround/config.lua#L277-L283

TL;DR - `vim.ui.input` works with cmdheight=0 and is what ought to be used to allow for other UI plugins to handle that input too.
